File: mycuration/src/main/java/com/phicdy/mycuration/view/SettingView.java
package com.phicdy.mycuration.view;

import android.support.annotation.NonNull;

public interface SettingView {
    void setUpdateInterval(int index, @NonNull String summary);
    void setAutoUpdateInMainUi(boolean isAutoUpdateInMainUi);
    void setArticleSort(boolean isNewArticleTop);
    void setInternalBrowser(boolean isEnabled);
    void setAllReadBehavior(int index, @NonNull String summary);
    void setSwipeDirection(int index, @NonNull String summary);
    void startLicenseActivity();
}

File: mycuration/src/main/java/com/phicdy/mycuration/view/SettingView.kt
package com.phicdy.mycuration.view

interface SettingView {
    fun setUpdateInterval(index: Int, summary: String)
    fun setAutoUpdateInMainUi(isAutoUpdateInMainUi: Boolean)
    fun setArticleSort(isNewArticleTop: Boolean)
    fun setInternalBrowser(isEnabled: Boolean)
    fun setAllReadBehavior(index: Int, summary: String)
    fun setSwipeDirection(index: Int, summary: String)
    fun startLicenseActivity()
}
